### PR TITLE
type: optimize CompoundedComponent type

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | type: optimize CompoundedComponent type |
| 🇨🇳 Chinese | type: optimize CompoundedComponent type |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/float-button/FloatButton.tsx [hunk 1]
```diff
@@ -107,10 +107,14 @@ const FloatButton = React.forwardRef<FloatButtonElement, FloatButtonProps>((prop
...
-}) as CompoundedComponent;
+});
+
+const FloatButton = InternalFloatButton as CompoundedComponent;
 
 if (process.env.NODE_ENV !== 'production') {
   FloatButton.displayName = 'FloatButton';
 }
 
+export { InternalFloatButton };
```
Comment (zombieJ) [inline components/float-button/FloatButton.tsx:118]: 这个不应该 export

File: components/float-button/interface.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import type React from 'react';
 import type { BadgeProps } from '../badge';
 import type { TooltipProps } from '../tooltip';
 import type BackTop from './BackTop';
+import type { InternalFloatButton } from './FloatButton';
```
Comment (zombieJ) [inline components/float-button/interface.ts:6]: 这么写两个文件定义相互绕圈，过于耦合了

Comment (li-jia-nan) [reply components/float-button/interface.ts:6]: 要不都放在 index 文件？

File: components/float-button/interface.ts [hunk 3]
```diff
@@ -68,10 +69,9 @@ export interface BackTopProps extends Omit<FloatButtonProps, 'target'> {
   duration?: number;
 }
 
-export type CompoundedComponent = React.ForwardRefExoticComponent<
-  FloatButtonProps & React.RefAttributes<FloatButtonElement>
-> & {
+export type CompoundedComponent = typeof InternalFloatButton & {
   Group: typeof Group;
   BackTop: typeof BackTop;
+  /** @internal */
```
Comment (zombieJ) [inline components/float-button/interface.ts:75]: 不要 internal，防止被 strip。React 里的 DO_NOT 也是放那儿的，可以用，但是开发者要自己承担后果。

File: components/mentions/demo/component-token.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { ConfigProvider, Mentions } from 'antd';
 
+// @ts-ignore
```
Comment (zombieJ) [inline components/mentions/demo/component-token.tsx:4]: 见上，不要 `internal`。`@ts-ignore` 这种非常脏，下同。

File: components/float-button/FloatButton.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/float-button/FloatButton.tsx]: done

File: components/float-button/interface.ts (hunk 2 above)
Comment (li-jia-nan) [reply components/float-button/interface.ts]: done

File: components/float-button/interface.ts (hunk 3 above)
Comment (li-jia-nan) [reply components/float-button/interface.ts]: done

File: components/mentions/demo/component-token.tsx (hunk 4 above)
Comment (li-jia-nan) [reply components/mentions/demo/component-token.tsx]: done

File: components/float-button/demo/render-panel.tsx [hunk 5]
```diff
@@ -2,7 +2,6 @@ import React from 'react';
 import { CustomerServiceOutlined, QuestionCircleOutlined, SyncOutlined } from '@ant-design/icons';
 import { FloatButton } from 'antd';
 
-/** Test usage. Do not use in your production. */
 const { _InternalPanelDoNotUseOrYouWillBeFired: InternalFloatButton } = FloatButton;
```
Comment (zombieJ) [inline components/float-button/demo/render-panel.tsx:5]: 不要删掉这个话

Comment (li-jia-nan) [reply components/float-button/demo/render-panel.tsx]: 误删了，我加回来了

**Review by zombieJ — APPROVED** [review]

**Pull request merged**